src/hooks/useArrowKeyFocusManager.ts

```diff
@@ -45,51 +52,134 @@ export default function useArrowKeyFocusManager({
 
     const arrowDownCallback = useCallback(() => {
         if (maxIndex < 0) {
             return;
         }
 
+        const nextIndex = disableCyclicTraversal ? maxIndex : 0;
+
         setFocusedIndex((actualIndex) => {
-            const currentFocusedIndex = actualIndex < maxIndex ? actualIndex + 1 : 0;
-            let newFocusedIndex = currentFocusedIndex;
+            let currentFocusedIndex = -1;
 
+            if (actualIndex === -1) {
+                currentFocusedIndex = 0;
+            } else if (allowHorizontalArrowKeys) {
+                currentFocusedIndex = actualIndex < maxIndex ? actualIndex + itemsPerRow : nextIndex;
+            } else {
+                currentFocusedIndex = actualIndex < maxIndex ? actualIndex + 1 : nextIndex;
+            }
+
+            if (disableCyclicTraversal && currentFocusedIndex > maxIndex) {
+                return actualIndex;
+            }
+
+            let newFocusedIndex = currentFocusedIndex;
             while (disabledIndexes.includes(newFocusedIndex)) {
-                newFocusedIndex = newFocusedIndex < maxIndex ? newFocusedIndex + 1 : 0;
+                if (actualIndex < 0) {
+                    newFocusedIndex += 1;
+                } else {
+                    newFocusedIndex += allowHorizontalArrowKeys ? itemsPerRow : 1;
+                }
+
+                if (newFocusedIndex < 0) {
+                    break;
+                }
                 if (newFocusedIndex === currentFocusedIndex) {
                     // all indexes are disabled
                     return actualIndex;
                 }
             }
-
             return newFocusedIndex;
         });
-    }, [disabledIndexes, maxIndex]);
+    }, [allowHorizontalArrowKeys, disableCyclicTraversal, disabledIndexes, itemsPerRow, maxIndex]);
     useKeyboardShortcut(CONST.KEYBOARD_SHORTCUTS.ARROW_DOWN, arrowDownCallback, arrowConfig);
 
+    const arrowLeftCallback = useCallback(() => {
+        if (maxIndex < 0) {
+            return;
+        }
+
+        const nextIndex = disableCyclicTraversal ? -1 : maxIndex;
+
+        setFocusedIndex((actualIndex) => {
+            let currentFocusedIndex = -1;
+            currentFocusedIndex = actualIndex > 0 ? actualIndex - 1 : nextIndex;
+
+            let newFocusedIndex = currentFocusedIndex;
+
+            while (disabledIndexes.includes(newFocusedIndex)) {
+                newFocusedIndex = newFocusedIndex > 0 ? newFocusedIndex - 1 : nextIndex;
+
+                if (newFocusedIndex === currentFocusedIndex) {
+                    // all indexes are disabled
+                    return actualIndex; // no-op
+                }
+            }
+            return newFocusedIndex;
+        });
+    }, [disableCyclicTraversal, disabledIndexes, maxIndex]);
+    useKeyboardShortcut(CONST.KEYBOARD_SHORTCUTS.ARROW_LEFT, allowHorizontalArrowKeys ? arrowLeftCallback : () => {}, arrowConfig);
```

Comment:
Does it makes sense to add an if into the `arrowLeftCallback` so it just return, because `useKeyboardShortcut` expect a stable callback

```
 if (maxIndex < 0) {
            return;
  }
   if (!allowHorizontalArrowKeys) {
            return;
    }
    ...
  ```
  then
  ```
  useKeyboardShortcut(CONST.KEYBOARD_SHORTCUTS.ARROW_LEFT, arrowLeftCallback, arrowConfig);
  ```
  you will need to add `allowHorizontalArrowKeys` as dependency for the useCallback